chromium_src/components/translate/core/browser/translate_manager.cc

```diff
@@ -5,9 +5,71 @@
 
 #include "components/translate/core/browser/translate_manager.h"
 
+#include "brave/components/translate/core/browser/brave_translate_language_filter.h"
+#include "components/translate/core/browser/translate_download_manager.h"
+#include "components/translate/core/browser/translate_prefs.h"
+
+namespace translate {
+
+class BraveIsSupportedTargetLanguageProxy : public TranslateDownloadManager {
+ public:
+  static bool IsSupportedLanguage(const std::string& lang) {
+    return IsTargetLanguageCodeSupported(lang);
+  }
+  ~BraveIsSupportedTargetLanguageProxy() override;
+};
+
+}  // namespace translate
+
+#define GetRecentTargetLanguage                                              \
+  GetRecentTargetLanguage();                                                 \
+  static_assert(base::StringPiece(__FUNCTION__).find("GetTargetLanguage") != \
```

Comment:
I think we can just use `base::StringPiece(__FUNCTION__) == "..."`, because `__FUNCTION__` builtin returns only function name.